Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift

```diff
@@ -139,16 +139,37 @@ public final class DefaultSettingsProvider: DefaultSettingsProviding {
             essentialKeys: DefaultSettingsProvider.essentialTargetSettings,
             newXcodeKeys: DefaultSettingsProvider.xcodeVersionSpecificSettings
         )
+        let mergeableSettings = mergeableSettings(for: target, configuration: buildConfiguration)
         var settings: SettingsDictionary = [:]
         settingsHelper.extend(buildSettings: &settings, with: targetDefaultAll)
         settingsHelper.extend(buildSettings: &settings, with: additionalTargetDefaults)
         settingsHelper.extend(buildSettings: &settings, with: targetDefaultVariant)
+        settingsHelper.extend(buildSettings: &settings, with: mergeableSettings)
         settingsHelper.extend(buildSettings: &settings, with: projectOverridableTargetDefaultSettings(for: project))
         return settings.filter(filter)
     }
 
     // MARK: - Private
 
+    private func mergeableSettings(for target: Target, configuration: BuildConfiguration) -> SettingsDictionary {
+        var settings: SettingsDictionary = [:]
+        if target.mergeable {
+            if target.product != .dynamicLibrary {
+                fatalError("Only products that are dynamic libraries can be marked as mergeable")
+            }
+            settings["MAKE_MERGEABLE"] = "YES"
+            settings["MERGEABLE_LIBRARY"] = "YES"
+        }
+        if target.mergedBinaryType == .automatic {
+            settings["MERGED_BINARY_TYPE"] = "automatic"
+        } else if case let .manual(mergeableDependencies) = target.mergedBinaryType {
+            settings["MERGED_BINARY_TYPE"] = "manual"
+            let frameworkLinkage = configuration.variant == .release ? "-merge_framework" : "-reexport_framework"
+            settings["OTHER_LDFLAGS"] = .array(mergeableDependencies.map { "-Wl,\(frameworkLinkage),\($0)" })
```

Comment:
❤️ This is neat! That way developers declare the dependencies and we take care of mapping them to the right build settings under the hood.